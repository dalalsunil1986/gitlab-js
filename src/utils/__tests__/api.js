jest.mock('../../config', () => ({
  privateToken: 'privateToken',
  userId: 'userId',
  apiVersion: '/api/v4',
  hostname: 'gitlab.com',
  port: 443,
}));

jest.mock('../request', () => jest.fn());

const api = require('../api');
const request = require('../request');


describe('api', () => {
  beforeEach(() => {
    request.mockReset();
    request.mockImplementation((...params) => [...params]);
  });


  it('Call api with url param only', async () => {
    const url = '/projects';

    const response = api(url);

    expect(request.mock.calls).toMatchSnapshot();
    expect(response).toMatchSnapshot();
  });


  it('Call api with url and params', async () => {
    const url = '/projects';
    const params = { state: 'opened' };

    const response = api(url, params);

    expect(request.mock.calls).toMatchSnapshot();
    expect(response).toMatchSnapshot();
  });


  it('Call api with url, params and opts', async () => {
    const url = '/projects';
    const params = { state: 'opened', order_by: 'order by' };
    const opts = { anotherOption: 'another option' };

    const response = api(url, params, opts);

    expect(request.mock.calls).toMatchSnapshot();
    expect(response).toMatchSnapshot();
  });


  it('POST', async () => {
    const url = '/projects';
    const params = { state: 'opened' };
    const opts = { method: 'POST' };

    const response = api(url, params, opts);

    expect(request.mock.calls).toMatchSnapshot();
    expect(response).toMatchSnapshot();
  });


  it('Throws an exception', () => {
    const url = '/projects';
    let error;
    request.mockReset();
    request.mockImplementation(() => {
      throw new Error('Api error');
    });

    try {
      api(url);
    } catch (err) {
      error = err;
    }

    expect(error).toMatchSnapshot();
  });
});
