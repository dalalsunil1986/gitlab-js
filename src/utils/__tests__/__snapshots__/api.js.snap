// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`api Call api with url and params 1`] = `
Array [
  Array [
    Object {
      "hostname": "gitlab.com",
      "method": "GET",
      "path": "/api/v4/projects?state=opened&private_token=privateToken",
      "port": 443,
    },
  ],
]
`;

exports[`api Call api with url and params 2`] = `
Array [
  Object {
    "hostname": "gitlab.com",
    "method": "GET",
    "path": "/api/v4/projects?state=opened&private_token=privateToken",
    "port": 443,
  },
]
`;

exports[`api Call api with url param only 1`] = `
Array [
  Array [
    Object {
      "hostname": "gitlab.com",
      "method": "GET",
      "path": "/api/v4/projects?private_token=privateToken",
      "port": 443,
    },
  ],
]
`;

exports[`api Call api with url param only 2`] = `
Array [
  Object {
    "hostname": "gitlab.com",
    "method": "GET",
    "path": "/api/v4/projects?private_token=privateToken",
    "port": 443,
  },
]
`;

exports[`api Call api with url, params and opts 1`] = `
Array [
  Array [
    Object {
      "anotherOption": "another option",
      "hostname": "gitlab.com",
      "method": "GET",
      "path": "/api/v4/projects?state=opened&order_by=order%20by&private_token=privateToken",
      "port": 443,
    },
  ],
]
`;

exports[`api Call api with url, params and opts 2`] = `
Array [
  Object {
    "anotherOption": "another option",
    "hostname": "gitlab.com",
    "method": "GET",
    "path": "/api/v4/projects?state=opened&order_by=order%20by&private_token=privateToken",
    "port": 443,
  },
]
`;

exports[`api POST 1`] = `
Array [
  Array [
    Object {
      "hostname": "gitlab.com",
      "method": "POST",
      "path": "/api/v4/projects?state=opened&private_token=privateToken",
      "port": 443,
    },
  ],
]
`;

exports[`api POST 2`] = `
Array [
  Object {
    "hostname": "gitlab.com",
    "method": "POST",
    "path": "/api/v4/projects?state=opened&private_token=privateToken",
    "port": 443,
  },
]
`;

exports[`api Throws an exception 1`] = `[Error: Api error]`;
